### PR TITLE
Feature udm user

### PR DESCRIPTION
**Pull request by keachi**
##### ISSUE TYPE
- New Module Pull Request
##### COMPONENT NAME

udm_user
##### ANSIBLE VERSION

```
ansible 2.2.0
```
##### SUMMARY

This module allows to manage users on a univention corporate server.
Example task:

```
- udm_user: name=foobar password=secure_password firstname=Foo lastname=Bar
```

To use this module, the target system must be an UCS. The module use the python API of an univention corporate server, called univention directory manager, to create, edit and delete users.
The arguments `position`, `ou` and `subpath` can be used to define the path in the ldap directory, this can be useful if the user shouldn't be at the default location.

This module needs the [PR #16172](https://github.com/ansible/ansible/pull/16172).

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @2-B for this new module. When this module receives 'shipit' comments from two community members and any 'needs_revision' comments have been resolved, we will mark for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

File: univention/udm_user.py [hunk 1]
```diff
@@ -0,0 +1,514 @@
...
+                                           type='str'),
+            mailPrimaryAddress      = dict(default=None,
+                                           type='str'),
+            mobileTelephoneNumber   = dict(default=[],
+                                           type='list'),
+            organisation            = dict(default=None,
+                                           type='str'),
+            pagerTelephonenumber    = dict(default=[],
+                                           type='list'),
+            password                = dict(default=None,
```
Comment (gundalow) [inline univention/udm_user.py:366]: For password / secret arguments `no_log=True` should be set

See http://docs.ansible.com/ansible/developing_modules.html#module-checklist for details

**Comment by keachi** [comment]
Thanks for reviewing. I changed that.

**Comment by gregdek** [comment]
Thanks @2-B for this PR. Unfortunately, it is not mergeable in its current state due to merge conflicts. Please rebase your PR. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

For help on how to do this cleanly please see http://docs.ansible.com/ansible/community.html#contributing-code-features-or-bugfixes

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@2-B A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by keachi** [comment]
ready_for_review

**Comment by gregdek** [comment]
Thanks @2-B for this new module. When this module receives 'shipit' comments from two community members and any 'needs_revision' comments have been resolved, we will mark for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
Thanks @2-B for this PR. Unfortunately, it is not mergeable in its current state due to merge conflicts. Please rebase your PR. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

For help on how to do this cleanly please see http://docs.ansible.com/ansible/community.html#contributing-code-features-or-bugfixes

[This message brought to you by your friendly Ansibull-bot.]

**Comment by keachi** [comment]
ready_for_review

**Comment by gregdek** [comment]
Thanks @2-B for this new module. When this module receives 'shipit' comments from two community members and any 'needs_revision' comments have been resolved, we will mark for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
Thanks @2-B for this PR. Unfortunately, it is not mergeable in its current state due to merge conflicts. Please rebase your PR. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

For help on how to do this cleanly please see http://docs.ansible.com/ansible/community.html#contributing-code-features-or-bugfixes

[This message brought to you by your friendly Ansibull-bot.]

**Comment by keachi** [comment]
ready_for_review

**Comment by gregdek** [comment]
Thanks @2-B for this new module. When this module receives 'shipit' comments from two community members and any 'needs_revision' comments have been resolved, we will mark for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gundalow** [comment]
shipit

**Comment by karras** [comment]
shipit

**Comment by gregdek** [comment]
Thanks again to @2-B for this PR, and thanks @ansible/core for reviewing. Marking for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gundalow** [comment]
Thanks for the new module.

**Pull request merged**